nixos/modules/services/backup/restic.nix

```
@@ -96,13 +96,22 @@ in
         };
 
         repository = mkOption {
-          type = types.str;
+          type = with types; nullOr str;
+          default = null;
           description = ''
             repository to backup to.
           '';
           example = "sftp:backup@192.168.1.100:/backups/${name}";
         };
 
+        repositoryFile = mkOption {
+          type = with types; nullOr path;
+          default = null;
+          description = ''
+            Path to the file the repository location to backup to.
```

This sounds unclear to me. Is the file content supposed to contain the repository location?

If so, just to make it clear:
```suggestion
            Path to the file containing the repository location to backup to.
```

---

Also, it might be worth testing this option in the restic test as well. Perhaps maybe a separate local repository, point to it with `repositoryFile`, and test that the same way the other repos get tested.

---

Done.